packages/quantic/force-app/main/default/lwc/quanticSearchBoxSuggestionsList/quanticSearchBoxSuggestionsList.js

```diff
@@ -87,26 +124,126 @@ export default class QuanticSearchBoxSuggestionsList extends LightningElement {
     }
   }
 
-  get suggestionsToRender() {
-    return this.suggestions?.map((s, i) => ({
-      ...s,
-      isSelected: this.selectionIndex === i,
-    }));
+  get allOptions() {
+    if (this.shouldDisplayRecentQueries) {
+      const options = [
+        ...this.recentQueriesThatStartWithCurrentQuery,
+        ...this.querySuggestionsNotInRecentQueries,
+      ]?.map((option, index) => ({
+        ...option,
+        key: index + 1,
+        isSelected: this.selectionIndex === index + 1,
+        containerCSSClass: `${optionCSSClass} ${
+          this.selectionIndex === index + 1 ? 'slds-has-focus' : ''
+        }`,
+        icon: option.isRecentQuery ? 'utility:clock' : 'utility:search',
+        clickHandler: () => {
+          this.handleSuggestionSelection(option.rawValue, option.isRecentQuery);
+        },
+      }));
+      return [
+        {
+          key: 0,
+          isSelected: this.selectionIndex === 0,
+          isClearRecentQueryButton: true,
+        },
+        ...options,
+      ].slice(0, this.maxNumberOfSuggestions + 1);
+    }
+    return this.querySuggestionsNotInRecentQueries
+      .map((option, index) => ({
+        ...option,
+        key: index,
+        isSelected: this.selectionIndex === index,
+        containerCSSClass: `${optionCSSClass} ${
+          this.selectionIndex === index ? 'slds-has-focus' : ''
+        }`,
+        icon: 'utility:search',
+        clickHandler: () => {
+          this.handleSuggestionSelection(option.rawValue);
+        },
+      }))
+      .slice(0, this.maxNumberOfSuggestions);
   }
 
-  emitSuggestionHighlighted() {
-    const highlightChangeEvent = new CustomEvent('highlightchange', {
-      detail: this.suggestions[this.selectionIndex],
+  /**
+   * Returns the query suggestions that are not already in the recent queries list.
+   */
+  get querySuggestionsNotInRecentQueries() {
+    return (
+      this.suggestions?.filter(
+        (suggestion) =>
+          !this.recentQueriesThatStartWithCurrentQuery.some(
+            (recentQuery) => recentQuery.rawValue === suggestion.rawValue
+          )
+      ) || []
+    );
+  }
+
+  /**
+   * Returns the recent queries that start with the query currently typed by the end user.
+   */
+  get recentQueriesThatStartWithCurrentQuery() {
```

Comment:
This should really be in Headless no?

We should be able to pass the current search query and headless already contains the state that has the recent queries, and it should return only the queries that starts with?

Reply:
Currently in Headless we have a state/controller for just recent queries, and not for recent queries that will be displayed as suggestions, thus currently in Headless there is no logic to filter out recent queries based on the current query.

It can be a nice improvement for Headless, but for now it is not supported.